### PR TITLE
DSI and DPI KMS fixes

### PR DESCRIPTION
**Pull request by 6by9**
One upstream DSI fix required for 5.15 only that sorts DSI probing.

The others could be backported if felt warranted, but aren't critical.
I'm largely working only on 5.15 now - shout if you'd prefer me to stay on 5.10.

### CODE REVIEW

File: arch/arm/configs/bcmrpi_defconfig [hunk 1]
```diff
@@ -896,6 +896,7 @@ CONFIG_FB_UDL=m
 CONFIG_FB_SIMPLE=y
 CONFIG_FB_SSD1307=m
 CONFIG_FB_RPISENSE=m
+CONFIG_BACKLIGHT_PWM=m
```
Comment (pelwell) [inline arch/arm/configs/bcmrpi_defconfig:899]: Please include the arm64 bcmrpi3_defconfig for completeness.

File: arch/arm/boot/dts/overlays/README [hunk 2]
```diff
@@ -3644,6 +3644,21 @@ Params: clock-frequency         Display clock frequency (Hz)
                                 value. NB also overridden by rgbXXX overrides.
         backlight-gpio          Defines a GPIO to be used for backlight control
                                 (default of none).
+        backlight-pwm           Defines a PWM channel to be used for backlight
+                                control (default of none). NB Disables audio
+                                headphone output as that also uses PWM.
+        backlight-pwm-chan      Choose channel on &pwm node for backlight control.
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/README]: You'll need to split this line.

File: arch/arm/boot/dts/overlays/README [hunk 3]
```diff
@@ -3644,6 +3644,21 @@ Params: clock-frequency         Display clock frequency (Hz)
...
+                                (default 0).
+        backlight-pwm-gpio      GPIO pin to be used for the PWM backlight. See
+                                pwm-2chan for valid options.
+                                (default 18 - note this can only work with
+                                 rgb666-padhi).
+        backlight-pwm-func      Function for GPIO pin used for the PWM backlight.
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/README]: And this one.

Comment (pelwell) [inline arch/arm/boot/dts/overlays/README]: "Pin function"?

**Comment by 6by9** [comment]
All done, although I forgot to add mention of the arm64 defconfig into the commit text. Are you fussed?

**Comment by pelwell** [comment]
> Are you fussed?

Do I look like an upstream dev?

I'll run the basic checks when not on a phone and merge it.

**Pull request merged**